vmdb/app/models/service_template.rb

```diff
@@ -180,4 +180,26 @@ def self.default_provisioning_entry_point
   def self.default_retirement_entry_point
     '/Service/Retirement/StateMachines/ServiceRetirement/default'
   end
+
+  def template_valid?
+    return true if service_resources.size.zero?
+
+    if atomic?
+      msg_prefix = "Service Template: [#{name}]:"
+      sr  = service_resources.first
+      obj = sr.resource.source
+
+      return "#{msg_prefix} Unable to find VM with Id [#{sr.resource.source_id}]" if obj.nil?
+      return "#{msg_prefix} VM/Template <#{obj.name}> with Id <#{obj.id}> is archived and cannot be used with provisioning." if obj.archived?
+      return "#{msg_prefix} VM/Template <#{obj.name}> with Id <#{obj.id}> is orphaned and cannot be used with provisioning." if obj.orphaned?
+      return true
+    end
+
+    message = nil
+    sub_services.each do |s|
+      message = s.template_valid?
+      break if message != true
+    end
+    message
```

Comment:
It is a UI code style that returns true if it is valid and otherwise returns a message that will be displayed on screen.